File: regexlib/2021-5-15/javascript_test_file/regexlib_2821.js
/* 2821
 * &quot;([^&quot;](?:\\.|[^\\&quot;]*)*)&quot;
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:"&quot;:"+"!"*16+"! _1_NQ"
 */
var REGEX = /&quot;([^&quot;](?:\\.|[^\\&quot;]*)*)&quot;/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '&quot;:' + '!'.repeat(i*1) + '! _1_NQ'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}

File: regexlib/python_re_test_file/regexlib_2821.py
# 2821
# &quot;([^&quot;](?:\\.|[^\\&quot;]*)*)&quot;
# EXPONENT
# nums:4
# EXPONENT AttackString:"&quot;:"+":"*16+"! _1_NQ"

import re
from time import perf_counter

regex = """&quot;([^&quot;](?:\\.|[^\\&quot;]*)*)&quot;"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "&quot;:" + ":" * i * 1 + "! _1_NQ"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")